collie/interactions/datasets.py

```diff
@@ -424,24 +430,30 @@ def __getitem__(self, index: Union[int, Iterable[int]]) -> (
 
         return (user_id, item_id), negative_ids_array
 
-    def _negative_sample(self, id: Union[int, np.array]) -> np.array:
-        """Generate negative samples for an ``id``."""
+    def _negative_sample(self, user_or_item_ids: Union[int, np.array]) -> np.array:
+        """Generate negative samples for ``user_or_item_ids``."""
         if self.negative_sample_type == 'item':
             num_ids = self.num_items
-            positive_set_tuple = '(specific_id, negative_id)'
+            while_loop_condition = (
+                lambda user_id, item_id: (user_id, item_id) in self.positive_set  # noqa: E731
+            )
```

Comment:
I don't know... I really love the `lambda` here, despite what `flake8` says.

Reply:
I agree! A whole function seems like overkill to me